### ISSUE TITLE
0.7.0 can not unlock older versions

### ISSUE THREAD

**Issue by mhogomchungu** [netheril96/securefs#24]
Below output is what i observe when i attempt to unlock a securefs volume created with one of the previous versions(i cant remember which one)
```
[mtz@ink ~]$ securefs mount  ../images/securefs/ sss
Password: 
[Warning] [0x7f1d027bb740] [2017-02-10 15:40:23.523422408 UTC]    Your filesystem is created by a vulnerable version of securefs.
Please immediate migrate your old data to a newly created securefs filesystem,
and remove all traces of old data to avoid information leakage!
Do you wish to continue with mounting? (y/n)[mtz@ink ~]$ 
```
As you can see,the question is asked by the application returns without waiting for a response. It also returns with zero indicating success when no such thing happened.

The behavior is there too when trying to run securefs as a daemon.

**Comment by netheril96** [comment]
Fixed in 2084193.

**Comment by mhogomchungu** [comment]
confirming the fix.

Also,can you remove all the warnings when building? 

sample output is below:
```
                                                      ^
/home/mtz/Downloads/securefs-master/external/cryptopp/oids.h:92:54: warning: extra ‘;’ [-Wpedantic]
     DEFINE_OID(certicom_ellipticCurve()+5, sect113r2);
                                                      ^
/home/mtz/Downloads/securefs-master/external/cryptopp/oids.h:93:55: warning: extra ‘;’ [-Wpedantic]
     DEFINE_OID(certicom_ellipticCurve()+15, sect163r2);
                                                       ^
/home/mtz/Downloads/securefs-master/external/cryptopp/oids.h:94:55: warning: extra ‘;’ [-Wpedantic]
     DEFINE_OID(certicom_ellipticCurve()+16, sect283k1);
                                                       ^
/home/mtz/Downloads/securefs-master/external/cryptopp/oids.h:95:55: warning: extra ‘;’ [-Wpedantic]
     DEFINE_OID(certicom_ellipticCurve()+17, sect283r1);
                                                       ^
/home/mtz/Downloads/securefs-master/external/cryptopp/oids.h:96:55: warning: extra ‘;’ [-Wpedantic]
     DEFINE_OID(certicom_ellipticCurve()+22, sect131r1);
                                                       ^
/home/mtz/Downloads/securefs-master/external/cryptopp/oids.h:97:55: warning: extra ‘;’ [-Wpedantic]
     DEFINE_OID(certicom_ellipticCurve()+23, sect131r2);
                                                       ^
/home/mtz/Downloads/securefs-master/external/cryptopp/oids.h:98:55: warning: extra ‘;’ [-Wpedantic]
     DEFINE_OID(certicom_ellipticCurve()+24, sect193r1);
                                                       ^
/home/mtz/Downloads/securefs-master/external/cryptopp/oids.h:99:55: warning: extra ‘;’ [-Wpedantic]
     DEFINE_OID(certicom_ellipticCurve()+25, sect193r2);
                                                       ^
/home/mtz/Downloads/securefs-master/external/cryptopp/oids.h:100:55: warning: extra ‘;’ [-Wpedantic]
     DEFINE_OID(certicom_ellipticCurve()+26, sect233k1);
                                                       ^
/home/mtz/Downloads/securefs-master/external/cryptopp/oids.h:101:55: warning: extra ‘;’ [-Wpedantic]
     DEFINE_OID(certicom_ellipticCurve()+27, sect233r1);
                                                       ^
/home/mtz/Downloads/securefs-master/external/cryptopp/oids.h:102:55: warning: extra ‘;’ [-Wpedantic]
     DEFINE_OID(certicom_ellipticCurve()+36, sect409k1);
                                                       ^
/home/mtz/Downloads/securefs-master/external/cryptopp/oids.h:103:55: warning: extra ‘;’ [-Wpedantic]
     DEFINE_OID(certicom_ellipticCurve()+37, sect409r1);
                                                       ^
/home/mtz/Downloads/securefs-master/external/cryptopp/oids.h:104:55: warning: extra ‘;’ [-Wpedantic]
     DEFINE_OID(certicom_ellipticCurve()+38, sect571k1);
                                                       ^
/home/mtz/Downloads/securefs-master/external/cryptopp/oids.h:105:55: warning: extra ‘;’ [-Wpedantic]
     DEFINE_OID(certicom_ellipticCurve()+39, sect571r1);
                                                       ^
[ 33%] Building CXX object external/cryptopp/CMakeFiles/cryptopp-object.dir/gost.cpp.o
[ 33%] Building CXX object external/cryptopp/CMakeFiles/cryptopp-object.dir/gzip.cpp.o
In file included from /home/mtz/Downloads/securefs-master/external/cryptopp/zdeflate.h:10:0,
                 from /home/mtz/Downloads/securefs-master/external/cryptopp/gzip.h:10,
                 from /home/mtz/Downloads/securefs-master/external/cryptopp/gzip.cpp:4:
/home/mtz/Downloads/securefs-master/external/cryptopp/filters.h:1000:64: warning: extra ‘;’ [-Wpedantic]
 DOCUMENTED_TYPEDEF(StringSinkTemplate<std::string>, StringSink);
                                                                ^
/home/mtz/Downloads/securefs-master/external/cryptopp/filters.h:1284:46: warning: extra ‘;’ [-Wpedantic]
 DOCUMENTED_TYPEDEF(StringSource, ArraySource);
                                              ^
[ 34%] Building CXX object external/cryptopp/CMakeFiles/cryptopp-object.dir/hex.cpp.o
In file included from /home/mtz/Downloads/securefs-master/external/cryptopp/basecode.h:10:0,
                 from /home/mtz/Downloads/securefs-master/external/cryptopp/hex.h:10,
                 from /home/mtz/Downloads/securefs-master/external/cryptopp/hex.cpp:7:
/home/mtz/Downloads/securefs-master/external/cryptopp/filters.h:1000:64: warning: extra ‘;’ [-Wpedantic]
 DOCUMENTED_TYPEDEF(StringSinkTemplate<std::string>, StringSink);
                                                                ^
/home/mtz/Downloads/securefs-master/external/cryptopp/filters.h:1284:46: warning: extra ‘;’ [-Wpedantic]
 DOCUMENTED_TYPEDEF(StringSource, ArraySource);
                                              ^
[ 35%] Building CXX object external/cryptopp/CMakeFiles/cryptopp-object.dir/hmac.cpp.o
[ 35%] Building CXX object external/cryptopp/CMakeFiles/cryptopp-object.dir/hrtimer.cpp.o
[ 36%] Building CXX object external/cryptopp/CMakeFiles/cryptopp-object.dir/ida.cpp.o
In file included from /home/mtz/Downloads/securefs-master/external/cryptopp/mqueue.h:6:0,
                 from /home/mtz/Downloads/securefs-master/external/cryptopp/ida.h:10,
                 from /home/mtz/Downloads/securefs-master/external/cryptopp/ida.cpp:6:
/home/mtz/Downloads/securefs-master/external/cryptopp/filters.h:1000:64: warning: extra ‘;’ [-Wpedantic]
 DOCUMENTED_TYPEDEF(StringSinkTemplate<std::string>, StringSink);
                                                                ^
/home/mtz/Downloads/securefs-master/external/cryptopp/filters.h:1284:46: warning: extra ‘;’ [-Wpedantic]
 DOCUMENTED_TYPEDEF(StringSource, ArraySource);
                                              ^
[ 37%] Building CXX object external/cryptopp/CMakeFiles/cryptopp-object.dir/idea.cpp.o
[ 37%] Building CXX object external/cryptopp/CMakeFiles/cryptopp-object.dir/iterhash.cpp.o
[ 38%] Building CXX object external/cryptopp/CMakeFiles/cryptopp-object.dir/keccak.cpp.o
[ 39%] Building CXX object external/cryptopp/CMakeFiles/cryptopp-object.dir/luc.cpp.o
In file included from /home/mtz/Downloads/securefs-master/external/cryptopp/pubkey.h:48:0,
                 from /home/mtz/Downloads/securefs-master/external/cryptopp/gfpcrypt.h:16,
                 from /home/mtz/Downloads/securefs-master/external/cryptopp/luc.h:8,
                 from /home/mtz/Downloads/securefs-master/external/cryptopp/luc.cpp:4:
/home/mtz/Downloads/securefs-master/external/cryptopp/filters.h:1000:64: warning: extra ‘;’ [-Wpedantic]
 DOCUMENTED_TYPEDEF(StringSinkTemplate<std::string>, StringSink);
                                                                ^
/home/mtz/Downloads/securefs-master/external/cryptopp/filters.h:1284:46: warning: extra ‘;’ [-Wpedantic]
 DOCUMENTED_TYPEDEF(StringSource, ArraySource);
                                              ^
[ 39%] Building CXX object external/cryptopp/CMakeFiles/cryptopp-object.dir/mars.cpp.o
[ 40%] Building CXX object external/cryptopp/CMakeFiles/cryptopp-object.dir/marss.cpp.o
[ 41%] Building CXX object external/cryptopp/CMakeFiles/cryptopp-object.dir/md2.cpp.o
[ 41%] Building CXX object external/cryptopp/CMakeFiles/cryptopp-object.dir/md4.cpp.o
[ 42%] Building CXX object external/cryptopp/CMakeFiles/cryptopp-object.dir/md5.cpp.o
[ 43%] Building CXX object external/cryptopp/CMakeFiles/cryptopp-object.dir/misc.cpp.o
[ 43%] Building CXX object external/cryptopp/CMakeFiles/cryptopp-object.dir/modes.cpp.o
[ 44%] Building CXX object external/cryptopp/CMakeFiles/cryptopp-object.dir/mqueue.cpp.o
In file included from /home/mtz/Downloads/securefs-master/external/cryptopp/mqueue.h:6:0,
                 from /home/mtz/Downloads/securefs-master/external/cryptopp/mqueue.cpp:7:
/home/mtz/Downloads/securefs-master/external/cryptopp/filters.h:1000:64: warning: extra ‘;’ [-Wpedantic]
 DOCUMENTED_TYPEDEF(StringSinkTemplate<std::string>, StringSink);
                                                                ^
/home/mtz/Downloads/securefs-master/external/cryptopp/filters.h:1284:46: warning: extra ‘;’ [-Wpedantic]
 DOCUMENTED_TYPEDEF(StringSource, ArraySource);
                                              ^
[ 45%] Building CXX object external/cryptopp/CMakeFiles/cryptopp-object.dir/mqv.cpp.o
In file included from /home/mtz/Downloads/securefs-master/external/cryptopp/pubkey.h:48:0,
                 from /home/mtz/Downloads/securefs-master/external/cryptopp/gfpcrypt.h:16,
                 from /home/mtz/Downloads/securefs-master/external/cryptopp/mqv.h:10,
                 from /home/mtz/Downloads/securefs-master/external/cryptopp/mqv.cpp:6:
/home/mtz/Downloads/securefs-master/external/cryptopp/filters.h:1000:64: warning: extra ‘;’ [-Wpedantic]
 DOCUMENTED_TYPEDEF(StringSinkTemplate<std::string>, StringSink);
                                                                ^
/home/mtz/Downloads/securefs-master/external/cryptopp/filters.h:1284:46: warning: extra ‘;’ [-Wpedantic]
 DOCUMENTED_TYPEDEF(StringSource, ArraySource);
                                              ^
[ 45%] Building CXX object external/cryptopp/CMakeFiles/cryptopp-object.dir/nbtheory.cpp.o
/home/mtz/Downloads/securefs-master/external/cryptopp/nbtheory.cpp:652:0: warning: ignoring #pragma omp parallel [-Wunknown-pragmas]
  #pragma omp parallel
 ^
/home/mtz/Downloads/securefs-master/external/cryptopp/nbtheory.cpp:653:0: warning: ignoring #pragma omp sections [-Wunknown-pragmas]
   #pragma omp sections
 ^
/home/mtz/Downloads/securefs-master/external/cryptopp/nbtheory.cpp:655:0: warning: ignoring #pragma omp section [-Wunknown-pragmas]
    #pragma omp section
 ^
/home/mtz/Downloads/securefs-master/external/cryptopp/nbtheory.cpp:657:0: warning: ignoring #pragma omp section [-Wunknown-pragmas]
    #pragma omp section
 ^
/home/mtz/Downloads/securefs-master/external/cryptopp/nbtheory.cpp:1004:0: warning: ignoring #pragma omp parallel [-Wunknown-pragmas]
  #pragma omp parallel
 ^
/home/mtz/Downloads/securefs-master/external/cryptopp/nbtheory.cpp:1005:0: warning: ignoring #pragma omp sections [-Wunknown-pragmas]
   #pragma omp sections
 ^
/home/mtz/Downloads/securefs-master/external/cryptopp/nbtheory.cpp:1007:0: warning: ignoring #pragma omp section [-Wunknown-pragmas]
    #pragma omp section
 ^
/home/mtz/Downloads/securefs-master/external/cryptopp/nbtheory.cpp:1012:0: warning: ignoring #pragma omp section [-Wunknown-pragmas]
    #pragma omp section
 ^
[ 46%] Building CXX object external/cryptopp/CMakeFiles/cryptopp-object.dir/network.cpp.o
In file included from /home/mtz/Downloads/securefs-master/external/cryptopp/network.h:8:0,
                 from /home/mtz/Downloads/securefs-master/external/cryptopp/network.cpp:5:
/home/mtz/Downloads/securefs-master/external/cryptopp/filters.h:1000:64: warning: extra ‘;’ [-Wpedantic]
 DOCUMENTED_TYPEDEF(StringSinkTemplate<std::string>, StringSink);
                                                                ^
/home/mtz/Downloads/securefs-master/external/cryptopp/filters.h:1284:46: warning: extra ‘;’ [-Wpedantic]
 DOCUMENTED_TYPEDEF(StringSource, ArraySource);
                                              ^
[ 47%] Building CXX object external/cryptopp/CMakeFiles/cryptopp-object.dir/oaep.cpp.o
In file included from /home/mtz/Downloads/securefs-master/external/cryptopp/pubkey.h:48:0,
                 from /home/mtz/Downloads/securefs-master/external/cryptopp/oaep.h:5,
                 from /home/mtz/Downloads/securefs-master/external/cryptopp/oaep.cpp:7:
/home/mtz/Downloads/securefs-master/external/cryptopp/filters.h:1000:64: warning: extra ‘;’ [-Wpedantic]
 DOCUMENTED_TYPEDEF(StringSinkTemplate<std::string>, StringSink);

```
